Subject: adding CERT_SUBJECT parameter to ext_auth plugin.

The subject field of the certificate is passed to the invoked script if
authentication mode is using certificates.

--- a/src/libcharon/plugins/ext_auth/ext_auth_listener.c
+++ b/src/libcharon/plugins/ext_auth/ext_auth_listener.c
@@ -98,6 +98,8 @@
 		process_t *process;
 		char *envp[32] = {};
 		int out, retval;
+		auth_cfg_t *auth;
+		certificate_t *cert;
 
 		*success = FALSE;
 
@@ -123,6 +125,13 @@
 					 ike_sa->get_other_eap_id(ike_sa));
 		}
 
+		auth = ike_sa->get_auth_cfg(ike_sa, TRUE);
+		cert = auth->get(auth, AUTH_HELPER_SUBJECT_CERT);
+		cert = auth->get(auth, AUTH_RULE_SUBJECT_CERT);
+		if (cert)
+			push_env(envp, countof(envp), "CERT_SUBJECT=%Y",
+					cert->get_subject(cert));
+
 		process = process_start_shell(envp, NULL, &out, NULL,
 									  "2>&1 %s", this->script);
 		if (process)
